optee-utee-build/src/linker.rs

```diff
@@ -126,10 +126,11 @@ impl Linker {
         out_dir: PathBuf,
         ta_dev_kit_dir: PathBuf,
     ) -> Result<(), Error> {
-        const ENV_TARGET_TA: &str = "TARGET_TA";
-        println!("cargo:rerun-if-env-changed={}", ENV_TARGET_TA);
+        // cargo passes TARGET as env to the build scripts
+        const ENV_TARGET: &str = "TARGET";
+        println!("cargo:rerun-if-env-changed={}", ENV_TARGET);
         let mut aarch64_flag = true;
-        match env::var(ENV_TARGET_TA) {
+        match env::var(ENV_TARGET) {
             Ok(ref v) if v == "arm-unknown-linux-gnueabihf" || v == "arm-unknown-optee" => {
```

Comment:
This function & inner logic is not straight forward and difficult to maintain.

Please leverage the Rust type system an use linear logic, instead of the boolean flag and nested if/else.
The convention here between src/ta.ld.S and created ta.lds is not straight forward.

It seems we can make it better by clearly handling Aarch64 and Arm32 separately and better leveraging the Type system to check the supported configurations.

Reply:
tracked in https://github.com/apache/teaclave-trustzone-sdk/issues/250